### PR TITLE
Trigger recreation of kustomization when helm values change

### PR DESCRIPTION
**Pull request by ianwesleyarmstrong**
Per https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/607, any changes in the values.yaml will cause the `kustomization` null_resource block to be tainted and recreated.

I'm not sure if there are other spots where the module level variables impact the values.yaml, but am happy to add those in (if there are any)!

### CODE REVIEW

**Comment by mysticaltech** [comment]
@ianwesleyarmstrong Simple and beautiful, we should have done it a long time ago. Thanks for taking care of it.

Between, at first sight, all the *_version variables can also be included, like `ccm_version` or `kured_version`, but there are others too.

When done, please make sure it's tried and tested! 🙏

**Comment by ianwesleyarmstrong** [comment]
> @ianwesleyarmstrong Simple and beautiful, we should have done it a long time ago. Thanks for taking care of it.
> 
> Between, at first sight, all the *_version variables can also be included, like `ccm_version` or `kured_version`, but there are others too.
> 
> When done, please make sure it's tried and tested! 🙏

Will test out once https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/623 is resolved!

**Comment by mysticaltech** [comment]
@ianwesleyarmstrong It's fixed! 🚀

File: init.tf [hunk 1]
```diff
@@ -86,6 +86,27 @@ resource "random_password" "rancher_bootstrap" {
...
+      local.longhorn_values,
+      local.cert_manager_values,
+      local.rancher_values,
+    ])
+    # Redeploy when versions of addons need to be updated
+    versions = join("\n", [
+      var.cluster_autoscaler_version,
+      var.hetzner_ccm_version,
+      var.hetzner_csi_version,
+      var.kured_version,
```
Comment (aleksasiriski) [inline init.tf:105]: This won't trigger when changing `var.kured_options`, I would suggest adding that as well.

**Review by aleksasiriski — APPROVED** [review]
Should be tested before merging, but seems okay to me.
A very welcome feature, thanks!

**Comment by ianwesleyarmstrong** [comment]
It appears, to be working! I noticed that it also triggered `null_resource.destroy_cluster_loadbalancer`during the plan. It doesn't actually change the `ingress_public_ipv{4/6}` outputs, but it could be annoying to see it diff everytime.

Before:
Screenshot 2023-03-07 at 5 52 56 PM
Terraform plan:
image
After:
image

**Comment by mysticaltech** [comment]
@ianwesleyarmstrong Good to hear. I will take care of null_resource.destroy_cluster_loadbalancer, I noticed it yesterday and it appears to be some legacy code not doing anything useful, as not successfully deleting the load balancer. If you can confirm that it's useless, feel free to remove it yourself.

Just one thing, why "N/A" in the coalesce values, it's just a random placeholder huh? If that's so, all good. I will wait for your final signal before merging.

**Comment by ianwesleyarmstrong** [comment]
> Just one thing, why "N/A" in the coalesce values, it's just a random placeholder huh? If that's so, all good. I will wait for your final signal before merging.

You are correct! Terraform didn't like the `join()` when there were nulls in the list (the default if a version isn't supplied), so it's just a way of keeping the compiler happy!

In that case, it should be ready to merge!

**Comment by ianwesleyarmstrong** [comment]
> Just one thing, why "N/A" in the coalesce values, it's just a random placeholder huh? If that's so, all good. I will wait for your final signal before merging.

You are correct! Terraform didn't like the `join()` when there were nulls in the list (the default if a version isn't supplied), so it's just a way of keeping the compiler happy!

In that case, it should be ready to merge!

**Comment by mysticaltech** [comment]
Perfect, will merge and release a new minor version later today! 🚀

**Comment by mysticaltech** [comment]
Released in v1.10.0 🚀 🙏

Also removed the useless `null_resource.destroy_cluster_loadbalancer`, proper logic to destroy the ingress LB and autoscaled nodes will come in the near future!

**Pull request merged**